include/mega/heartbeats.h

```diff
@@ -118,12 +88,71 @@ class HeartBeatBackupInfo : public CommandListener
     void updateLastActionTime();
 };
 
+/**
+ * @brief Backup info controlled by transfers
+ * progress is advanced with transfer progress
+ * by holding the count of total and transferred bytes
+ *
+ */
+class HeartBeatTransferProgressedInfo : public HeartBeatBackupInfo
+{
+public:
+    HeartBeatTransferProgressedInfo(handle backupId);
+    MEGA_DISABLE_COPY(HeartBeatTransferProgressedInfo)
+
+    virtual void updateTransferInfo(MegaTransfer *transfer);
+    virtual void removePendingTransfer(MegaTransfer *transfer);
+    virtual void clearFinshedTransfers();
+    virtual void setTotalBytes(long long value);
+    virtual void setTransferredBytes(long long value);
+
+    virtual double progress() const override;
+
+private:
+
+    long long mTotalBytes = 0;
+    long long mTransferredBytes = 0;
+
+
+    class PendingTransferInfo
+    {
+    public:
+        long long mTotalBytes = 0;
+        long long mTransferredBytes = 0;
+    };
+    std::map<int, std::unique_ptr<PendingTransferInfo>> mPendingTransfers;
+    std::vector<std::unique_ptr<PendingTransferInfo>> mFinishedTransfers;
+};
+
+
+class HeartBeatSyncInfo : public HeartBeatTransferProgressedInfo
+{
+public:
+    enum Status
+    {
+        UPTODATE = 1, // Up to date: local and remote paths are in sync
+        SYNCING = 2, // The sync engine is working, transfers are in progress
+        PENDING = 3, // The sync engine is working, e.g: scanning local folders
+        INACTIVE = 4, // Sync is not active. A state != ACTIVE should have been sent through '''sp'''
+        UNKNOWN = 5, // Unknown status
+    };
+
+    HeartBeatSyncInfo(int tag, handle backupId);
+    MEGA_DISABLE_COPY(HeartBeatSyncInfo)
+
+    virtual int syncTag() const;
```

Comment:
Are we planning to specialize this class? Otherwise, I think this method shouldn't be `virtual`

Reply:
not planning, but maybe onewaysyncs could potentially lead to a child class.